homestar-runtime/tests/utils.rs

```diff
@@ -0,0 +1,126 @@
+use anyhow::{Context, Result};
+use assert_cmd::{crate_name, prelude::*};
+#[cfg(not(windows))]
+use nix::{
+    sys::signal::{self, Signal},
+    unistd::Pid,
+};
+use once_cell::sync::Lazy;
+use retry::{delay::Fixed, retry};
+use std::net::{IpAddr, Ipv4Addr, Shutdown, SocketAddr, TcpStream};
+use std::{
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+use sysinfo::{PidExt, ProcessExt, ProcessRefreshKind, RefreshKind, SystemExt};
+
+static BIN: Lazy<PathBuf> = Lazy::new(|| assert_cmd::cargo::cargo_bin(crate_name!()));
+const IPFS: &str = "ipfs";
+
+pub fn startup_ipfs() -> Result<()> {
```

Comment:
guessing these can't be pub(crate)'ed? Just double-checkin (and pub(crate) in lib.rs too)

Reply:
Yeah, this totally worked out.